src/WebJobs.Extensions.Twilio/Bindings/TwilioSmsMessageAsyncCollector.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Extensions.Twilio;
+using Twilio;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Bindings
+{
+    internal class TwilioSMSMessageAsyncCollector : IAsyncCollector<SMSMessage>
+    {
+        private readonly TwilioSMSConfiguration _config;
+        private readonly TwilioSMSAttribute _attribute;
+        private readonly TwilioRestClient _client;
+
+        public TwilioSMSMessageAsyncCollector(TwilioSMSConfiguration config, TwilioSMSAttribute attribute, TwilioRestClient client)
+        {
+            _config = config;
+            _attribute = attribute;
+            _client = client;
+        }
+
+        public async Task AddAsync(SMSMessage item, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (item == null)
+            {
+                throw new ArgumentNullException("item");
+            }
+
+            ApplyDefaultMessageSettings(item, _attribute, _config);
+
+            if (string.IsNullOrEmpty(item.To))
+            {
+                throw new InvalidOperationException("A 'To' number must be specified for the message.");
+            }
+
+            if (string.IsNullOrEmpty(item.From))
+            {
+                throw new InvalidOperationException("A 'From' number must be specified for the message.");
+            }
+
+            if (string.IsNullOrEmpty(item.From))
+            {
+                throw new InvalidOperationException("A 'Body' must be specified for the message.");
+            }
+
+            var response = await Task.Run(() => _client.SendMessage(item.From, item.To, item.Body)).ConfigureAwait(false);
```

Comment:
> Why Task.Run?

Kicking off a Task for now since the current Twilio SDK does not expose async methods. I didn't want to block here as clients may have issues with that. Their new SDK (in alpha, but I might actually use here) does expose async methods, so this is temporary until we move to that. 

> Also, AsyncCollectors usually collect all the items and only send them all in the Flush method (see SendGrid and other bindings). Add is supposed to just queue them up.

This was initially implemented as you're describing, but I switched it to follow a model similar to the Mobile Apps and DocumentDB implementations and send the messages on add. I'll change that back.